src/Altinn.App.Core/Infrastructure/Clients/Storage/DataClient.cs

```diff
@@ -350,6 +351,30 @@ public async Task<DataElement> UpdateBinaryData(string org, string app, int inst
             _logger.LogError($"Updating attachment {dataGuid} for instance {instanceGuid} failed with status code {response.StatusCode}");
             throw await PlatformHttpException.CreateAsync(response);
         }
+        
+        /// <inheritdoc />
+        public async Task<DataElement> UpdateBinaryData(InstanceIdentifier instanceIdentifier, string? contentType, string filename, Guid dataGuid, Stream stream)
+        {
+            string apiUrl = $"{_platformSettings.ApiStorageEndpoint}instances/{instanceIdentifier}/data/{dataGuid}";
+            string token = JwtTokenUtil.GetTokenFromContext(_httpContextAccessor.HttpContext, _settings.RuntimeCookieName);
```

Comment:
Consider using IUserTokenProvider instead to avoid those HttpContext dependencies.